sw/device/silicon_creator/rom_ext/rescue.c

```diff
@@ -282,8 +303,20 @@ static rom_error_t protocol(rescue_state_t *state) {
   }
 }
 
-rom_error_t rescue_protocol(void) {
-  rom_error_t result = protocol(&rescue_state);
+rom_error_t rescue_protocol(boot_data_t *bootdata,
+                            const owner_rescue_config_t *config) {
+  rescue_state.config = config;
+  if ((hardened_bool_t)config == kHardenedBoolFalse) {
+    // If there is no rescue config, then the rescue region starts immediately
+    // after the ROM_EXT and ends at the end of the flash bank.
+    rescue_state.flash_start = CHIP_ROM_EXT_SIZE_MAX;
+    rescue_state.flash_limit = kFlashBankSize;
```

Comment:
This condition also seems good candidate for hardening, assuming the Owner may want to store critical data in the keep out area. The comparison itself is probably not subject to faults, but we may want to consider a redundant check in case a skip instruction can be mounted.

Reply:
Done.